### PR TITLE
Fix/print with invalid sources

### PR DESCRIPTION
**Pull request by inacht**
Ticket https://github.com/mapbender/mapbender/issues/853. The print pdf request is made via xhttp instead of via form submitting now (refactoring). A new tab is not opened anymore.  Invalid layer urls and internal server side errors do not hinder the generation of the pdf document.

### CODE REVIEW

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.printClient.js [hunk 1]
```diff
@@ -513,9 +513,34 @@
...
+                        document.body.appendChild(link);
+
+                        link.click();
+
+                        setTimeout(function(){ document.body.removeChild(link); }, 2000);
+                    }
+                };
+
+                request.send(form.serialize());
             }
```
Comment (werrolf) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.printClient.js]: Why?  
This is completely unneccessary and unrelated. It also disables HTML5 form validation.

File: src/Mapbender/PrintBundle/Component/PrintService.php [hunk 2]
```diff
@@ -271,27 +271,29 @@ private function getImages($width, $height)
...
+            try {
+
+                $mapRequestResponse = $this->mapRequest($url);
+            } catch (\Exception $e) {
+                continue;
+            }
 
             $imageName    = $this->makeTempFile('mb_print');
-            $imageNames[] = $imageName;
             $rawImage = $this->serviceResponseToGdImage($imageName, $mapRequestResponse);
```
Comment (werrolf) [inline src/Mapbender/PrintBundle/Component/PrintService.php]: You disabled temp file cleanup for images that downloaded but could not be opened by GD. Why?

File: src/Mapbender/PrintBundle/Component/PrintService.php [hunk 3]
```diff
@@ -271,27 +271,29 @@ private function getImages($width, $height)
...
                 $logger->debug("ERROR! PrintRequest failed: " . $url);
                 $logger->debug($mapRequestResponse->getContent());
-                print_r('an error has occurred. see log for more details <br>');
-                print_r($mapRequestResponse->getContent());
-                foreach ($imageNames as $i => $imageName) {
-                    unlink($imageName);
-                }
-                exit;
+                continue;
             }
```
Comment (werrolf) [inline src/Mapbender/PrintBundle/Component/PrintService.php]: This is the only change we need.

**Comment by inacht** [comment]
@werrolf ok, i don't mind, if you just merge the lines, where you think they are needed.

**Comment by werrolf** [comment]
[Other pull](https://github.com/mapbender/mapbender/pull/1013) merged after approval, I'm closing this one.  

Branch is scheduled for deletion within ~a week.

**Pull request closed without merging**